### DISCUSSION TITLE
Storybook build is failing in Vite app using "@storybook/builder-vite": "^0.2.2",

### DISCUSSION THREAD

**Discussion by satyendera** [storybookjs/builder-vite#481] [General]
When we build storybook build facing this error 
[vite]: Rollup failed to resolve import "window" from "node_modules/@storybook/channel-websocket/dist/esm/index.js".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
`build.rollupOptions.external`
transforming (2341) node_modules/@storybook/addon-measure/node_modules/core-js/internals/regexp-flags.jsinfo => Manager built (17 s)
ERR! Error: [vite]: Rollup failed to resolve import "window" from "node_modules/@storybook/channel-websocket/dist/esm/index.js".
<img width="1184" alt="Screen Shot 2022-08-16 at 2 50 05 PM" src="https://user-images.githubusercontent.com/6051914/184800173-80f29c1b-37cc-4087-bfda-da07192ebdda.png">

**Comment by IanVS** [comment]
Hi, what version of storybook are you using?  Can you share your `.storybook/main.js`?

**Reply by satyendera** [reply]
module.exports = {
  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "@storybook/addon-interactions",
    
  ],
  core: {
    builder: "@storybook/builder-vite",
  },
  features: { storyStoreV7: true },
  framework: "@storybook/react",
  stories: ["../src/**/*.stories.@(js|jsx|ts|tsx)", "../src/**/*.stories.mdx"],
  async viteFinal(config, { configType }) {
    console.log('configTypeconfigTypeconfigType',configType);
    if (configType === "DEVELOPMENT") {
      
    }

    config.define = {
      ...config.define,
      global: "window",
      
    };

    return config;
  },
};


versions are 
"@storybook/builder-vite": "^0.2.2",
    "@storybook/react": "^6.5.10",

**Reply by satyendera** [reply]
this is fixed by adding  "window": "^4.2.7" in dev dependancies

**Comment by ShaneyWaris** [comment]
@IanVS Found this error during the build with vite. Any idea what may cause this issue? 
<img width="890" alt="image" src="https://user-images.githubusercontent.com/48903475/186149947-7449abf0-b7c2-4edc-a3ab-93c5010a9dd5.png">

**Reply by IanVS** [reply]
This looks like a build command for your app, not for storybook.  And it looks like you're missing an index.html file that vite needs as the entry point.

**Reply by satyendera** [reply]
yea , this error is coming from Vite config , not related to storybook

**Reply by IanVS** [reply]
I think you'll have better luck asking in the vite repo then, this repo is specifically for running vite in storybook.

**Reply by IanVS** [reply]
But, from what it looks like, you don't have an `index.html` in your project root, or you've configured your project root incorrectly so that vite can't find the file you do have.
